File: api-java-sql/src/main/java/com/nebulahotel/controller/DashboardController.java
package com.nebulahotel.controller;

import com.nebulahotel.repository.HospedagensRepository;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/dashboard")
public class DashboardController {

    private final HospedagensRepository hospedagensRepository;

    public DashboardController(HospedagensRepository hospedagensRepository) {
        this.hospedagensRepository = hospedagensRepository;
    }

    @GetMapping("/faturamento")
    public Map<String, Object> faturamento() {
        return Map.of(
                "totalHospedagens", hospedagensRepository.count(),
                "faturamentoTotal", hospedagensRepository.calcularFaturamentoTotal()
        );
    }
}

File: api-kotlin-nosql/src/main/kotlin/com/nebulahotel/controller/DashboardController.kt
package com.nebulahotel.controller

import com.nebulahotel.repository.HospedagensRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.math.BigDecimal

@CrossOrigin
@RestController
@RequestMapping("/dashboard")
class DashboardController(
    private val hospedagensRepository: HospedagensRepository
) {

    @GetMapping("/faturamento")
    fun faturamento(): Map<String, Any> {
        val hospedagens = hospedagensRepository.findAll()
        val total = hospedagens.fold(BigDecimal.ZERO) { acc, item -> acc + item.totalGeral }

        return mapOf(
            "totalHospedagens" to hospedagens.size,
            "faturamentoTotal" to total
        )
    }
}
